### PR TITLE
Simplification of SampleToken

### PR DESCRIPTION
**Pull request by realphant0m**
Fixes #

**ERC20.sol**
1. `decreaseAllowance`: I suggest changing the parameter naming of the int `subtractedValue` to just `value`. During the function call we are only dealing with one value, the function name is already a clear indication of what this parameter will be used for. 
2. `increaseAllowance`: I suggest changing the parameter naming of the int `addedValue` to just `value`: During the function call we are only dealing with one value, the function name is already a clear indication of what this parameter will be used for. 

**ERC20Detailed.sol**

3. `_decimals`: I suggest changing the variable visibility of the `_decimals` variable from `private` to `internal` and the reasoning being: This contract is supposed to be inherited by future children contracts, good practice is to reuse already stored values in favour of magic numbers or added storage duplication. The current implementation requires children contracts to duplicate the `_decimals` variable to utilise it for i.e. mathematical operations. 

**SimpleToken.sol**

4. `DECIMALS`: I suggest removing the `DECIMALS` variable and the reasoning being: We already inherit a `_decimals` variable from **ERC20Detailed.sol**. We can simply change the visibility of the `_decimals` variable inherited from **ERC20Detailed.sol** from `private` to `internal` which results in improved simplicity and contract-storage management. 

5. `INITIAL_SUPPLY`: I suggest removing the `INITIAL_SUPPLY` variable and the reasoning being: This is supposed to be a simple token example, one could argue that introducing an `INITIAL_SUPPLY` variable in addition to an inherited `totalSupply` variable goes against the nature of being a simple token. The `INITIAL_SUPPLY `variable is only used as an argument input to the `_mint` function during contract initialisation and it stores the same value as `totalSupply` thus removing the `INITIAL_SUPPLY` variable would result in improved simplicity.

### CODE REVIEW

**Comment by nventuro** [comment]
Hello @realphant0m, thanks for contributing to OpenZeppelin! A couple comments about your suggestions:

>ERC20.sol
> decreaseAllowance
> increaseAllowance

A simple `value` name may imply that the allowance is being decreased/increased _to_ that value: the current names make explicit the fact that the passed value will be added/subtracted.

>_decimals: I suggest changing the variable visibility of the _decimals variable from private to internal and the reasoning being: This contract is supposed to be inherited by future children contracts, good practice is to reuse already stored values in favour of magic numbers or added storage duplication. The current implementation requires children contracts to duplicate the _decimals variable to utilise it for i.e. mathematical operations.

The `_decimals` variable can be accessed via the `decimals()` function. If we made the variable `internal`, child contracts could modify it at will _after_ construction.

>SimpleToken.sol

This is an example contract: the intent of those constants is to make clear what they represent (e.g. to make the `ERC20Detailed` constructor call less cryptic), not for child contracts to read from them.

**Comment by realphant0m** [comment]
@nventuro I agree with you mostly, how I believe we should ditch the duplicated DECIMALS value from the simpleToken.sol contract, we could just initialise the constructor with a function call to the private _decimals value and set the initial supply during initialisation?

```
contract SimpleToken is ERC20, ERC20Detailed {
    uint256 public initialSupply; 

    constructor () public ERC20Detailed("SimpleToken", "SIM", decimals()) {
        initialSupply = 10000 * (10 ** uint256(decimals()));
        _mint(msg.sender, initialSupply);
    }

```

Does this make it more cryptic? this way we are still doing the same but cheaper?

**Comment by frangio** [comment]
@realphant0m I think removing the `DECIMALS` constant is a good idea. Can you change this PR to include that change only? Thanks!

**Comment by realphant0m** [comment]
@frangio Yes I will have that done by the end of the day. Cheers.

File: contracts/examples/SimpleToken.sol [hunk 1]
```diff
@@ -10,13 +10,11 @@ import "../token/ERC20/ERC20Detailed.sol";
...
-    uint8 public constant DECIMALS = 18;
-    uint256 public constant INITIAL_SUPPLY = 10000 * (10 ** uint256(DECIMALS));
 
     /**
      * @dev Constructor that gives msg.sender all of existing tokens.
      */
-    constructor () public ERC20Detailed("SimpleToken", "SIM", DECIMALS) {
-        _mint(msg.sender, INITIAL_SUPPLY);
+    constructor () public ERC20Detailed("SimpleToken", "SIM", 18) {
+        _mint(msg.sender, 10000 * (10 ** uint256(_decimals));
```
Comment (frangio) [inline contracts/examples/SimpleToken.sol]: `_decimals` is a private field.

```suggestion
        _mint(msg.sender, 10000 * (10 ** uint256(decimals()));
```

**Pull request merged**